aiogram/dispatcher/dispatcher.py

```diff
@@ -1030,3 +1030,67 @@ def _wrap_async_task(self, callback, run_task=None) -> callable:
         if run_task:
             return self.async_task(callback)
         return callback
+
+    def throttled(self, on_throttled: typing.Optional[typing.Callable] = None,
+                  key=None, rate=None,
+                  user_id=None, chat_id=None):
+        """
+        Meta-decorator for throttling.
+        Invokes on_throttled if the handler was throttled.
+
+        Example:
+
+        .. code-block:: python3
+            async def handler_throttled(message: types.Message, **kwargs):
+                message.answer("Throttled!")
+
+            @dp.throttled(handler_throttled)
+            async def some_handler(message: types.Message )
+
+        :param on_throttled: the callable object that should be either a function or return a coroutine
+        :param key: key in storage
+        :param rate: limit (by default is equal to default rate limit)
+        :param user_id: user id
+        :param chat_id: chat id
+        :return: decorator
+        """
+
+        def decorator(func):
+            nonlocal on_throttled, key, rate, user_id, chat_id
```

Comment:
![](https://telegra.ph/file/8bb4b763a78ee7840ed45.jpg)

Reply:
Oh yes, it's useless definition, thank you. Will update PR soon

Reply:
So, I checked this again. The reason why I do it this way because the decorator itself must have only one argument and it's the function to be wrapped, while here I need a bunch of args to configure throttling

Reply:
There’re mutables for storing something outside functions in situations like this

Reply:
Refactored in [4d00b7c](https://github.com/aiogram/aiogram/pull/181/commits/4d00b7c50242612f8289bca4d86d621f428ff741)

Reply:
I supposed it's resolved now